frontend/src/metabase/query_builder/components/view/View.jsx

```diff
@@ -422,13 +431,43 @@ class View extends Component {
     const { isNative } = Lib.queryDisplayInfo(question.query());
 
     const isNewQuestion = !isNative && Lib.sourceTableOrCardId(query) === null;
-    const isModelOrMetric =
-      question.type() === "model" || question.type() === "metric";
+    const isModel = question.type() === "model";
+    const isMetric = question.type() === "metric";
 
-    if (isModelOrMetric && queryBuilderMode === "dataset") {
+    if ((isModel || isMetric) && queryBuilderMode === "dataset") {
```

Comment:
This `queryBuilderMode === "dataset"` is a bit confusing here. Maybe it's time to rename `DatasetEditor` to `ModelEditor`?